src/main/java/com/homihq/db2rest/rest/rpc/SubRoutine.java

```diff
@@ -0,0 +1,44 @@
+package com.homihq.db2rest.rest.rpc;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.jdbc.core.simple.SimpleJdbcCall;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Map;
+import java.util.Objects;
+@RequiredArgsConstructor
+@Slf4j
+public abstract class SubRoutine {
+
+    final JdbcTemplate jdbcTemplate;
+
+    Map<String, Object> execute(String subRoutineName, Map<String, Object> inParams) throws SQLException {
+        validate(subRoutineName);
+        return callSubRoutine(subRoutineName, inParams);
+    }
+
+    Map<String, Object> callSubRoutine(String subRoutineName, Map<String, Object> inParams) {
+        jdbcTemplate.setResultsMapCaseInsensitive(true);
+        SqlParameterSource in = new MapSqlParameterSource()
+                .addValues(inParams);
+        return getSimpleJdbcCall(subRoutineName).execute(in);
+    }
+
+    boolean isInvalidSubroutine(String subRoutineName) throws SQLException {
+        try(var connection = Objects.requireNonNull(jdbcTemplate.getDataSource()).getConnection()) {
```

Comment:
What happens to the connection and resultset object? are they closed automatically?